common/server_player.cpp

```diff
@@ -1987,14 +1987,20 @@ Response::ResponseCode Server_Player::cmdChangeZoneProperties(const Command_Chan
 
         ges.enqueueGameEvent(event, playerId);
 
-        if (!zone->getCards().isEmpty() && cmd.always_reveal_top_card()) {
-            Event_RevealCards revealEvent;
-            revealEvent.set_zone_name(zone->getName().toStdString());
-            revealEvent.set_card_id(0);
-            zone->getCards().first()->getInfo(revealEvent.add_cards());
+        revealTopCardIfNeeded(zone, ges);
 
-            ges.enqueueGameEvent(revealEvent, playerId);
+        return Response::RespOk;
+    } else if (cmd.has_always_look_at_top_card()) {
```

Comment:
I don't see a problem with toggling both at the same time, the event should be queued only once of course.

Reply:
In practice they will never true in the same event, since each checkbox sends it's own event with only one field filled in.

Reply:
it doesn't matter what the client does, if it is in the protocol it should be accounted for. we are soon to have more than one client.

Reply:
Updated, single event will be send even if both are set,